templates/tpus.libsonnet

```diff
@@ -18,9 +18,10 @@ local base = import 'base.libsonnet';
   TpuSpec:: base.BaseAccelerator {
     local tpu = self,
 
-    name: 'v%(version)d-%(size)d' % tpu,
+    name: 'v%(versionName)s-%(size)d' % tpu,
     type: 'tpu',
     version: error 'Must override `version`',
+    versionName: error 'Must override `versionName`',
```

Comment:
Can this default to `str(self.version)` for the common case?

Reply:
Alternatively, could we just change the type of `version` to a string instead of adding a new field? It looks like it's only used in this file.

Reply:
Probably not, due to this line `numCores: if tpu.version <= 3 then 8 else 4`.

Reply:
Hmm, it seems the number is being used in pytorch. https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fml-testing-accelerators%20numCores&type=code

Reply:
Being able to check version ranges is helpful for e.g. finding how many devices there are.

What about adding another field `variant`? Then `name` would be like `v%(version)d%(variant)s-%(size)d`, e.g. `v4-8` or `v4abcd-8`

Reply:
I like this one, and will update it.

Reply:
Tested it via http://shortn/_kvfc52UbDD. Also, tested internally.